ai_sdlc/services/context7_service.py

```
@@ -29,27 +28,25 @@ def __init__(self, cache_dir: Path):
+        except portalocker.exceptions.LockException as e:
+            raise TimeoutError("Could not acquire cache lock") from e
 
     def _release_lock(self) -> None:
         """Release the file lock."""
         try:
-            with open(self.cache_lock_file, 'r+') as lock_file:
-                fcntl.flock(lock_file.fileno(), fcntl.LOCK_UN)
+            if getattr(self, "_lock_handle", None) is not None:
+                self._lock_handle.release()
+                self._lock_handle = None
         except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)